src/shared/sleep-config.c

```diff
@@ -165,8 +165,30 @@ int can_sleep_disk(char **types) {
 
 #define HIBERNATION_SWAP_THRESHOLD 0.98
 
+/* entry in /proc/swaps */
+typedef struct SwapEntry {
+        char *device;
+        char *type;
+        uint64_t size;
+        uint64_t used;
+        int priority;
+} SwapEntry;
+
+static SwapEntry* swap_entry_free(SwapEntry *se) {
+        if (se) {
+                free(se->device);
+                free(se->type);
+                mfree(se);
```

Comment:
the whole point of mfree() is that it returns NULL, so that you can do:

```c
p = mfree(p);
```

if you don't care about the return value here, then a regular `free()` should be just fine